File: core/mill-metadata-core/src/main/java/io/qpointz/mill/metadata/configuration/MetadataProperties.java
package io.qpointz.mill.metadata.configuration;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Configuration properties for faceted metadata service (v2).
 * Uses separate config section to avoid conflicts with legacy metadata config.
 */
@Data
@ConfigurationProperties(prefix = "mill.metadata.v2")
public class MetadataProperties {
    
    private Storage storage = new Storage();
    private File file = new File();
    
    @Data
    public static class Storage {
        private String type = "file";  // file, jpa, composite, external
    }
    
    @Data
    public static class File {
        private String path = "classpath:metadata/example.yml";
        private boolean watch = false;  // Auto-reload on changes
    }
}

File: metadata/mill-metadata-autoconfigure/src/main/kotlin/io/qpointz/mill/metadata/configuration/MetadataProperties.kt
package io.qpointz.mill.metadata.configuration

import org.springframework.boot.context.properties.ConfigurationProperties

/** External configuration binding for metadata v2 storage setup. */
@ConfigurationProperties(prefix = "mill.metadata.v2")
class MetadataProperties {
    var storage: Storage = Storage()
    var file: File = File()

    /** Storage backend selector. */
    class Storage {
        var type: String = "file"
    }

    /** File-storage specific configuration. */
    class File {
        var path: String = "classpath:metadata/example.yml"
        var watch: Boolean = false
    }
}
